remote-method/index.js

```diff
@@ -0,0 +1,285 @@
+'use strict';
+var yeoman = require('yeoman-generator');
+var chalk = require('chalk');
+
+var actions = require('../lib/actions');
+var helpers = require('../lib/helpers');
+var validateName = helpers.validateName;
+var typeChoices =helpers.getTypeChoices;
+var accepts = [];
+var returns = [];
+var https = [];
+
+module.exports = yeoman.generators.Base.extend({
+  // This generator does not track file changes via yeoman,
+  // as loopback-workspace is editing (modifying) files when
+  // saving project changes.
+  constructor: function() {
+    yeoman.generators.Base.apply(this, arguments);
+
+    this.argument('modelName', {
+      desc: 'Name of the model',
+      required: false,
+      type: String
+    });
+  },
+
+  help: function() {
+    return helpers.customHelp(this);
+  },
+
+  loadProject: actions.loadProject,
+
+  loadModels: actions.loadModels,
+
+  askForModel: function() {
+    if (this.modelName) {
+      this.modelDefinition = this.projectModels.filter(function(m) {
+        return m.name === this.modelName;
+      }.bind(this))[0];
+
+      if(!this.modelDefinition) {
+        var msg = 'Model not found: ' + this.modelName +
+                  ' Please choose from Model List:';
+        this.log(chalk.red(msg));
+      }
+    }
+
+    if (!this.modelDefinition) {
+      var done = this.async();
+      var prompts = [
+        {
+          name: 'model',
+          message: 'Select the model:',
+          type: 'list',
+          choices: this.modelNames
+        }
+      ];
+      this.prompt(prompts, function(answers) {
+        this.modelName = answers.model;
+        done();
+      }.bind(this));
+    }
+  },
+
+  findModelDefinition: function() {
+    this.modelDefinition = this.projectModels.filter(function(m) {
+        return m.name === this.modelName;
+    }.bind(this))[0];
+
+    if (!this.modelDefinition) {
+        var msg = 'Model not found: ' + this.modelName;
+        this.log(chalk.red(msg));
+        this.async()(new Error(msg));
```

Comment:
> hmm...actually I use this.async() to keep it consistent with other files, eg:
https://github.com/strongloop/generator-loopback/blob/master/relation/index.js#L54
https://github.com/strongloop/generator-loopback/blob/master/property/index.js#L56

Ah, I didn't realise that, let's keep your current solution then! Consistency is important.

> And I tried throw new Error(msg) it has the same performance, so that means somewhere in yeoman catches that error. I will check yeoman doc to confirm that. 
> And could you please explain it more like why throw is simpler than this.async ? Both of them work me but I am a little confused with their difference (in terms of speed, etc), thank you so much.

I guess this is subjective. As I read the code, it's switching the action to an async one, only to immediately and synchronously end the action with an error. When reading code like this, it forces me to stop and re-read the line again to understand what's actually happening there, it's not immediately obvious on the first sigh (to me).

OTOH, `try new Error(msg)` is very clear (to me).

The micro-performance of `this.async(err)` may be better than `throw err`, because IIRC v8 does not optimise methods containing exception handling code (although I am not sure if it applies to `try/catch` blocks only or whether `throw` is affected too). However, in this particular case, the impact on runtime performance is negligible.

Anyways, as I wrote at the top, let's stay consistent with other parts of this codebase.